BotConfigurations/BotConfiguration.cs

```diff
@@ -0,0 +1,16 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace WeatherMonitoringAndReportingService.BotConfigurations
+{
+    public class BotConfiguration
+    {
+        public bool Enabled { get; set; }
+        public int? HumidityThreshold { get; set; }
+        public int? TemperatureThreshold { get; set; }
+        public string Message { get; set; }
+    }
```

Comment:
this project should be built by follow Open-Close principle, which the model should be open to extend/ close for modification, But in your code it doesn't,
First: this model should be more generic, since if I wanna add new Bot then I must modify the Model so I broke the principle,
How to fix that? you should make this as base class and be something like

```suggestion
    public class BotConfiguration
    {
        public bool Enabled { get; set; }
        public int? Threshold { get; set; }
        public string Message { get; set; }
    }
```
and each Bot type should inherit the base and override the Threshold by using JsonProperty attribute